### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by Dmole** [fikovnik/ShiftIt#188]
Sometimes shiftit will show the Spinning Beach Ball  on mouseover of the icon after using a key command to move a window, then shiftit will quit. the key command will work after re-opening shift it.

Process:               ShiftIt [63355]
Path:                  /Applications/ShiftIt.app/Contents/MacOS/ShiftIt
Identifier:            org.shiftitapp.ShiftIt
Version:               1.6.3 (1.6.3)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           ShiftIt [63355]
User ID:               501

Date/Time:             2015-11-23 09:46:37.752 -0500
OS Version:            Mac OS X 10.10.5 (14F1021)
Report Version:        11
Anonymous UUID:        9B05D90F-8820-3005-3775-BA1FE637354D

Sleep/Wake UUID:       7D108531-EAD9-4618-B1AA-D6BED9EC41CA

Time Awake Since Boot: 2700000 seconds
Time Since Wake:       650000 seconds

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       EXC_I386_GPFLT

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   org.shiftitapp.ShiftIt        	0x000000010f41417a __63-[X11WindowDriver(WindowDelegate) getGeometry_:ofWindow:error:]_block_invoke + 54
1   org.shiftitapp.ShiftIt        	0x000000010f414129 -[X11WindowDriver(WindowDelegate) getGeometry_:ofWindow:error:] + 118
2   org.shiftitapp.ShiftIt        	0x000000010f4130da -[X11Window getGeometry:screen:error:] + 78
3   org.shiftitapp.ShiftIt        	0x000000010f423d15 -[AbstractWindowGeometryShiftItAction execute:error:] + 536
4   org.shiftitapp.ShiftIt        	0x000000010f4194fe -[SIWindowManager executeAction:error:] + 228
5   org.shiftitapp.ShiftIt        	0x000000010f417bcb -[ShiftItAppDelegate invokeShiftItActionByIdentifier_:] + 510
6   org.shiftitapp.ShiftIt        	0x000000010f41c44b hotKeyHandler + 177
7   com.apple.HIToolbox           	0x00007fff8787fb6c DispatchEventToHandlers(EventTargetRec*, OpaqueEventRef*, HandlerCallRec*) + 1260
8   com.apple.HIToolbox           	0x00007fff8787efae SendEventToEventTargetInternal(OpaqueEventRef*, OpaqueEventTargetRef*, HandlerCallRec*) + 386
9   com.apple.HIToolbox           	0x00007fff8787ee22 SendEventToEventTargetWithOptions + 43
10  com.apple.HIToolbox           	0x00007fff878bba9e ToolboxEventDispatcherHandler(OpaqueEventHandlerCallRef*, OpaqueEventRef*, void*) + 1762
11  com.apple.HIToolbox           	0x00007fff87880295 DispatchEventToHandlers(EventTargetRec*, OpaqueEventRef*, HandlerCallRec*) + 3093
12  com.apple.HIToolbox           	0x00007fff8787efae SendEventToEventTargetInternal(OpaqueEventRef*, OpaqueEventTargetRef*, HandlerCallRec*) + 386
13  com.apple.HIToolbox           	0x00007fff87894cb6 SendEventToEventTarget + 40
14  com.apple.AppKit              	0x00007fff88d4f082 -[NSApplication sendEvent:] + 2104
15  com.apple.AppKit              	0x00007fff88c78b68 -[NSApplication run] + 711
16  com.apple.AppKit              	0x00007fff88bf5244 NSApplicationMain + 1832
17  libdyld.dylib                 	0x00007fff8c1235c9 start + 1

**Comment by fikovnik** [comment]
Do you use X11?

**Comment by Dmole** [comment]
Yes, I sometimes use Inkscape, but I think X11 was not running at the time or that crash, only Terminal, Chrome, and Eclipse. I am not able to reproduce this issue while running Inkscape in X11. ShiftIt
 will normally crash when the system is under load and I ask it to move a window.

**Comment by Dmole** [comment]
Today I was doing some really heavy disk IO so I tested again and was able to crash it 3 times, every time shift it erroneously opened X11.

**Comment by fikovnik** [comment]
For me it is difficult to reproduce, but there is definitely a bug :-/ The thing is that the usability API is not have very high fidelity and sometimes it does not correctly report back the window handler. In that case ShiftIt, incorrectly, fall backs to X11.

**Comment by Dmole** [comment]
Try using a HDD (not an SSD) and a disk testing tool to reproduce, I was using rdiff-backup and I could get it to crash quite reliably. .... maybe add an option to not try x11 and just give up if nothing is returned in a timely manner.

**Comment by cjcjameson** [comment]
Blocking 1.6.4 stable? I don't see a re-occurence of this from the past year +

**Comment by Dmole** [comment]
Oh I can reproduce it any time,
and accidentally do sometimes unintentionally.
Needs an option to skip X11 and give up gracefully.

**Comment by Dmole** [comment]
Still an issue in 1.6.6
